spring-subscription-callback/src/main/java/com/apollographql/subscription/callback/SubscriptionCallbackHandler.java

```diff
@@ -0,0 +1,201 @@
+package com.apollographql.subscription.callback;
+
+import com.apollographql.subscription.exception.InactiveSubscriptionException;
+import com.apollographql.subscription.message.CallbackMessageCheck;
+import com.apollographql.subscription.message.CallbackMessageComplete;
+import com.apollographql.subscription.message.CallbackMessageNext;
+import com.apollographql.subscription.message.SubscritionCallbackMessage;
+import graphql.ExecutionResult;
+import graphql.GraphqlErrorBuilder;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jetbrains.annotations.NotNull;
+import org.reactivestreams.Publisher;
+import org.springframework.graphql.server.WebGraphQlHandler;
+import org.springframework.graphql.server.WebGraphQlRequest;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+/** GraphQL subscription handler implementing Apollo HTTP callback protocol. */
+public class SubscriptionCallbackHandler {
+  private static final Log logger = LogFactory.getLog(SubscriptionCallbackHandler.class);
+
+  public static final String SUBSCRIPTION_PROTOCOL_HEADER = "subscription-protocol";
+  public static final String SUBSCRIPTION_PROTOCOL_HEADER_VALUE = "callback/1.0";
+
+  private final WebGraphQlHandler graphQlHandler;
+  private final Scheduler scheduler;
+
+  public SubscriptionCallbackHandler(WebGraphQlHandler graphQlHandler) {
+    this(graphQlHandler, Schedulers.boundedElastic());
+  }
+
+  public SubscriptionCallbackHandler(WebGraphQlHandler graphQlHandler, Scheduler scheduler) {
+    this.graphQlHandler = graphQlHandler;
+    this.scheduler = scheduler;
+  }
+
+  @NotNull
+  public Mono<Boolean> handleSubscriptionUsingCallback(
+      @NotNull WebGraphQlRequest graphQlRequest, @NotNull SubscriptionCallback callback) {
+    if (logger.isDebugEnabled()) {
```

Comment:
Do you really need to wrap this in `isDebugEnabled`? Shouldn't `logger.debug` do this check for you?

Reply:
In general yes as otherwise you end up creating the log string regardless whether you will be logging it or not (I guess you could rewrite log message to use a string template to avoid this but in general it is a good practice so we don't forget it in the future).